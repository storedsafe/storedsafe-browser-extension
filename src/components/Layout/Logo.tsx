import React from 'react';
import logo from './logo.png';
import './Logo.scss';

export const Logo: React.FunctionComponent = () => (
  <img src={logo} alt="StoredSafe" className="logo" />
);
