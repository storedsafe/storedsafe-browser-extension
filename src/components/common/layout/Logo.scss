@import '../../../styles/variables.scss';

.logo {
  display: block;
  max-height: 100%;
}
