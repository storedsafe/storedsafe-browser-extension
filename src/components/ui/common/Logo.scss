@import '../../../variables';

.logo {
  display: block;
  max-height: 100%;
}
