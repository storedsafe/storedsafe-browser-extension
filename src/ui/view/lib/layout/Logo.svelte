<script>
  import { getMessage, LocalizedMessage } from "../../../../global/i18n";
</script>

<style>
  img {
    max-width: 90%;
    max-height: 2.2em;
    margin: auto;
    user-select: none;
  }
</style>

<img src="/assets/logo.svg" alt={getMessage(LocalizedMessage.IMG_LOGO_ALT)} />
