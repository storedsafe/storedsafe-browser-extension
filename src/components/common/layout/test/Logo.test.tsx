import * as React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import { Logo } from '../Logo'
import pretty from 'pretty'

/**
 * Set up and tear down container to mount tested component in.
 * */
let container: Element = null
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div')
  document.body.appendChild(container)
})
afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

test('<Logo />', () => {
  act(() => {
    render(<Logo />, container)
  })

  expect(pretty(container.innerHTML)).toMatchSnapshot()
})
