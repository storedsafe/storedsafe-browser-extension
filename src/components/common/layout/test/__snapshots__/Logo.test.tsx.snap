// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> 1`] = `"<img src=\\"test-file-stub\\" alt=\\"StoredSafe\\" class=\\"logo\\">"`;
